actions/push/action.yaml

```diff
@@ -0,0 +1,168 @@
+    value: ${{ steps.diff.outputs.pushed }}
+  digest:
+    description: The digest of the OCI artifact. Empty if the artifact was not pushed.
+    value: ${{ steps.push.outputs.digest }}
+  digest-url:
+    description: The digest URL of the OCI artifact, i.e. including the repository URL. Empty if the artifact was not pushed.
+    value: ${{ steps.push.outputs.digest-url }}
+runs:
+  using: composite
+  steps:
+  - name: Check if the Flux CLI is installed
+    shell: bash
+    run: |
+      if ! flux version --client > /dev/null 2>&1; then
+        echo ""
+        echo "The Flux CLI is not installed. Please install it first using this GitHub Action:"
+        echo ""
+        echo "  https://github.com/controlplaneio-fluxcd/distribution/tree/main/actions/setup"
+        echo ""
+        exit 1
+      fi
+
+  - name: Check if the repository is valid
+    shell: bash
+    env:
+      REPOSITORY: ${{ inputs.repository }}
+    run: |
+      if [ "$REPOSITORY" != "$(echo $REPOSITORY | awk '{print tolower($1)}')" ]; then
+        echo ""
+        echo "The repository $REPOSITORY contains uppercase letters."
+        echo "OCI repositories must be lowercase, all the OCI tooling fails if they're not."
+        echo ""
+        exit 1
+      fi
+
+  - name: Define the --ignore-paths flag
+    id: ignore-paths
+    shell: bash
+    env:
+      IGNORE_PATHS: ${{ inputs.ignore-paths }}
+    run: |
+      ignore_paths_flag=""
+      if [ -n "$IGNORE_PATHS" ]; then
+        ignore_paths_flag="--ignore-paths=$IGNORE_PATHS"
+      fi
+
+      echo "flag=$ignore_paths_flag" >> $GITHUB_OUTPUT
+
+  - name: Diff the OCI artifact
+    id: diff
+    shell: bash
+    env:
+      REPOSITORY: ${{ inputs.repository }}
+      IGNORE_PATHS_FLAG: ${{ steps.ignore-paths.outputs.flag }}
+      INPUT_PATH: ${{ inputs.path }}
+      DIFF_TAG: ${{ inputs.diff-tag }}
+    run: |
+      pushed=true
+      if [ -n "$DIFF_TAG" ]; then
+        artifact_url=${REPOSITORY}:${DIFF_TAG}
+        if flux diff artifact oci://$artifact_url --path=$INPUT_PATH $IGNORE_PATHS_FLAG; then
+          pushed=false
+        fi
+      fi
+
+      echo "pushed=$pushed" >> $GITHUB_OUTPUT
+
+  - name: Push the OCI artifact
+    id: push
+    if: steps.diff.outputs.pushed == 'true'
+    shell: bash
+    env:
+      REPOSITORY: ${{ inputs.repository }}
+      IGNORE_PATHS_FLAG: ${{ steps.ignore-paths.outputs.flag }}
+      INPUT_PATH: ${{ inputs.path }}
+      DIFF_TAG: ${{ inputs.diff-tag }}
+      TAGS: ${{ inputs.tags }}
+    run: |
+      artifact_url=$REPOSITORY
+      if [ -n "$DIFF_TAG" ]; then
+        artifact_url=${REPOSITORY}:${DIFF_TAG}
+      elif [ -n "$TAGS" ]; then
+        IFS=',' read -ra tag_list <<< $TAGS
+        artifact_url=${REPOSITORY}:${tag_list[0]}
+      fi
+
+      echo "Pushing $INPUT_PATH to ${artifact_url}..."
+
+      push_output=$(flux push artifact oci://$artifact_url \
+        --source="$(git config --get remote.origin.url)" \
```

Comment:
This would work for GHCR, we need to set it to the HTTPS URL for GitHub to link it to the repo like this https://github.com/controlplaneio-fluxcd/flux-operator/blob/cc6818a7e89ea2825ef5fa718b0da5fe94947dc5/.github/workflows/push-manifests.yml#L50

Maybe we should allow the source URL as input.

Reply:
Yes, it works:

```json
  "labels": {
    "org.opencontainers.image.revision": "main@sha1:9cf989cad1e7ef94256c681e4b9733dd3197d063",
    "org.opencontainers.image.source": "https://github.com/matheuscscp/d1-fleet",
    "org.opencontainers.image.created": "2025-02-28T17:25:35Z"
  }
```

But I added `source` as a parameter anyway and added an example. GitHub Actions clones the repo with HTTPS, not SSH.